okhttp/src/main/java/io/grpc/okhttp/OkHttpClientTransport.java

```diff
@@ -414,6 +433,10 @@ public void close() {}
           executor.execute(clientFrameHandler);
         }
 
+        if (deadline != null) {
+          clientFrameHandler.startDeadlineWatchdog(deadline);
```

Comment:
This needs to happen earlier. connect() returns only after connect has completed. Ditto for TLS.